src/main/java/com/example/tradingapp/messaging/OkxCancelOrderListener.java

```diff
@@ -0,0 +1,74 @@
+package com.example.tradingapp.messaging;
+
+import com.example.tradingapp.strategy.CancelOrderPolicy;
+import com.example.tradingapp.tracker.Order;
+import com.example.tradingapp.trading.OkxOrderTracker;
+import com.example.tradingapp.trading.model.request.OkxCancelOrderRequest;
+import com.example.tradingapp.trading.model.request.OrderRequest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.stereotype.Component;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.TextMessage;
+import java.io.IOException;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class OkxCancelOrderListener {
+
+    private final CancelOrderPolicy cancelOrderPolicy;
+    private final OkxOrderTracker orderTracker;
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @JmsListener(destination = "order/cancel/okx")
+    public void handle(Message message) throws IOException, JMSException {
+
+        if (message instanceof TextMessage textMessage) {
+            try {
+                OrderRequest orderRequest = mapper.readValue(textMessage.getText(), OrderRequest.class);
+                log.info("Cancel Order Request from Solace: {}", orderRequest);
+
+                var order = buildOrderFromOrderRequest(orderRequest);
+
+                boolean foundOrderToCancel = false;
+                for (var entity : orderTracker.getPlacedOrders().entrySet()) {
```

Comment:
We can improve this by creating a method in `OkxOrderTracker` that will try to find an order
The signature will look something like

```
Optional<Order> findMatchingOrder(Order order);
```

It will return `Optional.empty()` if the order is not present
So we can get rid of `foundOrderToCancel` flag